akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala

```diff
@@ -383,16 +352,67 @@ trait RejectionHandlerFactory {
   def create(): RejectionHandler
 }
 
+trait QueueSelectorFactory {
+  def create(): QueueSelector
+}
+
+/**
+ * A `QueueSelector` is responsible for, given a `Runnable` and the number of available
+ * queues, return which of the queues that `Runnable` should be placed in.
+ */
+trait QueueSelector {
+  /**
+   * Must be deterministic—return the same value for the same input.
```

Comment:
Might be too general for the specific case. But returning different queues for the same runnable wouldn't be very much of an Affinity tho.